File: 628._Maximum_Product_of_Three_Numbers.py
class Solution:
    def maximumProduct(self, nums):
        nums.sort()
        # print(nums)
        if(nums[0]<0 and nums[1]<0):
            # print(nums[0]*nums[1]*nums[-1])
            # print(nums[-1]*nums[-2]*nums[-3])
            if(nums[0]*nums[1]*nums[-1]>nums[-1]*nums[-2]*nums[-3]):
                return nums[0]*nums[1]*nums[-1]
            else:
                return nums[-1]*nums[-2]*nums[-3]
        else:
            return nums[-1]*nums[-2]*nums[-3]

File: 628._Maximum_Product_of_Three_Numbers.cpp
class Solution {
public:
    int maximumProduct(vector<int>& nums) {
        std::sort(nums.begin(),nums.end());
        if(nums[0]<0 && nums[1]<0){
            if(nums[0]*nums[1]*nums[nums.size()-1]>nums[nums.size()-1]*nums[nums.size()-2]*nums[nums.size()-3])
                return nums[0]*nums[1]*nums[nums.size()-1];
            else
                return nums[nums.size()-1]*nums[nums.size()-2]*nums[nums.size()-3];               
        }
        return nums[nums.size()-1]*nums[nums.size()-2]*nums[nums.size()-3];
    }
};